src/test/java/uk/gov/ch/developer/docs/controller/developer/SignOutControllerTest.java

```diff
@@ -0,0 +1,113 @@
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class SignOutControllerTest {
+
+    @Mock
+    private IIdentityProvider identityProviders;
+    @Mock
+    private Store store;
+    @Mock
+    private HttpServletRequest httpServletRequest;
+    @Mock
+    private HttpServletResponse httpServletResponse;
+    @Mock
+    private Session session;
+    private Map<String, Object> data;
+
+    @InjectMocks
+    private SignOutController signOutController;
+
+    @Test
+    @DisplayName("Test that a valid signed in user's session state is correctly altered")
+    public void testDoSignOutWhenSignedIn() throws IOException {
+        final String zxsValue = "0000000001z";
+        SignInInfo signInInfo = new SignInInfo();
+        signInInfo.setSignedIn(true);
+
+        when(httpServletRequest.getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY))
+                .thenReturn(session);
+        when(session.getSignInInfo()).thenReturn(signInInfo);
+        when(session.getData()).thenReturn(setUserSessionData(zxsValue));
+
+        signOutController.doSignOut(httpServletResponse, httpServletRequest);
+
+        assertFalse(data.containsKey(SessionKeys.SIGN_IN_INFO.getKey()));
+
+        verify(store).delete(zxsValue);
+        verify(httpServletResponse).sendRedirect(identityProviders.getRedirectUriPage());
+    }
+
+    @Test
+    @DisplayName("Test that a user not signed in is unable to alter session state and is correctly redirected back home")
+    public void testDoSignOutWhenUserIsNotSignedIn() throws IOException {
+        SignInInfo signInInfo = new SignInInfo();
+        signInInfo.setSignedIn(false);
+
+        when(httpServletRequest.getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY))
+                .thenReturn(session);
+        when(session.getSignInInfo()).thenReturn(signInInfo);
+
+        signOutController.doSignOut(httpServletResponse, httpServletRequest);
+
+        verify(session, times(0)).getData();
+        verify(httpServletResponse).sendRedirect(identityProviders.getRedirectUriPage());
+    }
+
+    @Test
+    @DisplayName("Test that a valid signed in user's without a valid cookie is unable to delete from store")
+    public void testSignOutWhenUserHasInvalidZXSKey() throws IOException {
+        final String zxsValue = null;
+        SignInInfo signInInfo = new SignInInfo();
+        signInInfo.setSignedIn(true);
+
+        when(httpServletRequest.getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY))
+                .thenReturn(session);
+        when(session.getSignInInfo()).thenReturn(signInInfo);
+        when(session.getData()).thenReturn(setUserSessionData(zxsValue));
+
+        signOutController.doSignOut(httpServletResponse, httpServletRequest);
+
+        assertFalse(data.containsKey(SessionKeys.SIGN_IN_INFO.getKey()));
+
+        verify(store, times(0)).delete(zxsValue);
+        verify(httpServletResponse).sendRedirect(identityProviders.getRedirectUriPage());
+    }
+
```

Comment:
Please don't rely on side effects of this being called. 
It is much clearer to be explicit.

In a test, I would suggest that your structure is on the lines of:
```
private Map<String, Object> data=setUserSessionData(...);
...
when(session.getData()).thenReturn(data);
...
assert...(data...);

```